staging/src/k8s.io/client-go/plugin/pkg/client/auth/oidc/oidc_test.go

```diff
@@ -150,6 +150,82 @@ func TestClientCache(t *testing.T) {
 	assertCacheLen(t, cache, 2)
 }
 
+func TestNewClient(t *testing.T) {
+	tests := []struct {
+		name         string
+		config       map[string]string
+		wantErr      string
+		wantProvider bool
```

Comment:
```suggestion
```

This is not required because we don't need the provider nil/not-nil check. The callers will check if err is nil/not-nil and if it's not-nil they're not going to use the `provider`.